sympy/core/mul.py

```diff
@@ -195,6 +195,10 @@ def flatten(cls, seq):
                                 bargs.insert(0, ar)
                             bargs = [Add._from_args(bargs)]
                             rv = bargs, [], None
+            if a.is_commutative == True and b.is_commutative == True:
+                if ((a.is_finite == False and b.is_zero == True) or
+                        (b.is_finite == False and a.is_zero == True)):
+                    rv = [S.NaN], [], None
```

Comment:
You are right. I was just being more verbose. `== False` is required though, otherwise, it will fail to handle the case of `None`.